src/tmx.js

```diff
@@ -537,16 +538,18 @@ class TMX {
             if (attrs.srclang) {
                 this.sourceLocale = attrs.srclang;
             }
+            if (attrs.adminlang) {
+                this.adminLocale = attrs.adminlang;
+            }
         }
         if (tmx.body) {
             if (tmx.body.tu) {
                 const units = makeArray(tmx.body.tu);
                 for (let i = 0; i < units.length; i++) {
                     const unit = units[i];
                     const tu = new TranslationUnit();
-                    if (unit._attributes & unit._attributes.srclang) {
-                        tu.sourceLocale = unit._attributes.srclang;
-                    }
+                    tu.sourceLocale = (unit._attributes && unit._attributes.srclang) || this.sourceLocale || this.adminLocale || "en";
```

Comment:
is there some reason to put the last default locale is "en" instead of "en-US"?

Reply:
tmx files are often by language instead of by whole locale, so the default should be "en". In the next commit, I changed the default to "en" only